### ISSUE TITLE
prefer-screen-queries: false positives when query used from within

### ISSUE THREAD

**Issue by Belco90** [testing-library/eslint-plugin-testing-library#116]
Rule `prefer-screen-queries` is throwing false positives when a query is called from `within` util:

```javascript
import { render, within, screen } from 'react-testing-library';

// ...
render(<SomeComponent />);
const section = screen.getByTestId('some-id');

// this is fine but the rule complains about it.
const submit = within(section).getByText(/submit/i);
```

**Comment by Belco90** [comment]
@all-contributors please add @Belco90 for bugs
